apps/antalmanac/src/actions/AppStoreActions.ts

```diff
@@ -102,6 +102,33 @@ export const saveSchedule = async (userID: string, rememberMe: boolean) => {
     }
 };
 
+export async function autoSaveSchedule(userID: string) {
+    logAnalytics({
+        category: analyticsEnum.nav.title,
+        action: analyticsEnum.nav.actions.SAVE_SCHEDULE,
+        label: userID,
+    });
+    if (userID != null) {
+        userID = userID.replace(/\s+/g, '');
+
+        if (userID.length > 0) {
+            const scheduleSaveState = AppStore.schedule.getScheduleAsSaveState();
```

Comment:
You can apply guard clauses here to reduce nesting.

```suggestion
    if (userID == null) return
    userID = userID.replace(/\s+/g, '');

    if (userID.length < 0) return
    const scheduleSaveState = AppStore.schedule.getScheduleAsSaveState();
```